src/mds/ScrubHeader.h

```diff
@@ -19,14 +19,17 @@
 #include <memory>
 #include <string>
 #include <string_view>
+#include <condition_variable>
```

Comment:
This line should not be necessary. `common/fair_mutex.h` picks that up from `common/ceph_mutex.h`.

I'm not sure why, even so, you get this:

```
FAILED: src/crimson/CMakeFiles/crimson-common.dir/__/msg/Message.cc.o 
/usr/bin/ccache /usr/bin/clang++-12  -DBOOST_ALL_NO_LIB -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_ASIO_DISABLE_THREAD_KEYWORD_EXTENSION -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_RANDOM_DYN_LINK -DBOOST_THREAD_DYN_LINK -DCEPH_INSTALL_DATADIR=\"/usr/local/share/ceph\" -DCEPH_INSTALL_FULL_PKGLIBDIR=\"/usr/local/lib/ceph\" -DCMAKE_INSTALL_LIBDIR=\"lib\" -DHAVE_CONFIG_H -DSEASTAR_API_LEVEL=6 -DSEASTAR_DEBUG -DSEASTAR_DEBUG_SHARED_PTR -DSEASTAR_DEFAULT_ALLOCATOR -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -DSEASTAR_SHUFFLE_TASK_QUEUE -DSEASTAR_TYPE_ERASE_MORE -DWITH_SEASTAR=1 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_REENTRANT -D_THREAD_SAFE -D__CEPH__ -D__STDC_FORMAT_MACROS -D__linux__ -Isrc/include -I../src -I../src/seastar/include -Isrc/seastar/gen/include -isystem /opt/ceph/include -isystem include -isystem ../src/xxHash -isystem ../src/rapidjson/include -isystem ../src/jaegertracing/opentelemetry-cpp/api/include -isystem ../src/jaegertracing/opentelemetry-cpp/exporters/jaeger/include -isystem ../src/jaegertracing/opentelemetry-cpp/ext/include -isystem ../src/jaegertracing/opentelemetry-cpp/sdk/include -g -Werror -fPIC   -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -Wall -fno-strict-aliasing -fsigned-char -Wtype-limits -Wignored-qualifiers -Wpointer-arith -Werror=format-security -Winit-self -Wno-unknown-pragmas -Wnon-virtual-dtor -Wno-ignored-qualifiers -ftemplate-depth-1024 -Wpessimizing-move -Wredundant-move -Wno-inconsistent-missing-override -Wno-mismatched-tags -Wno-unused-private-field -Wno-address-of-packed-member -Wno-unused-function -Wno-unused-local-typedef -Wno-varargs -Wno-gnu-designator -Wno-missing-braces -Wno-parentheses -Wno-deprecated-register -DCEPH_DEBUG_MUTEX -D_GLIBCXX_ASSERTIONS -fdiagnostics-color=auto -Wno-non-virtual-dtor -std=gnu++17 -U_FORTIFY_SOURCE -DSEASTAR_SSTRING -Wno-error=unused-result "-Wno-error=#warnings" -fstack-clash-protection -fsanitize=address -fsanitize=undefined -fno-sanitize=vptr -std=c++17 -MD -MT src/crimson/CMakeFiles/crimson-common.dir/__/msg/Message.cc.o -MF src/crimson/CMakeFiles/crimson-common.dir/__/msg/Message.cc.o.d -o src/crimson/CMakeFiles/crimson-common.dir/__/msg/Message.cc.o -c ../src/msg/Message.cc
In file included from ../src/msg/Message.cc:141:
In file included from ../src/messages/MMDSCacheRejoin.h:20:
In file included from ../src/mds/CInode.h:37:
In file included from ../src/mds/CDentry.h:34:
In file included from ../src/mds/ScrubHeader.h:25:
```

`common/ceph_mutex.h` should be defining `ceph::condition_variable` here:

https://github.com/ceph/ceph/blob/dab3364f179167df026128a0d51a1dbe52d1f820/src/common/ceph_mutex.h#L80

I think you'll need to look at the preprocessor output to debug this.

Reply:
I can see that `-DWITH_SEASTAR=1` has been defined on the build command-line; and looking at `src/common/ceph_mutex.h`, it is obvious that `condition_variable` is not defined when `WITH_SEASTAR` is deifned

So, if Jenkins builds are now scheduled to always build with `WITH_SEASTAR=1`, then `ceph_mutex.h` needs to define `condition_variable` appropriately for the Seastar case.